source/containers/document-pii-detection/parsers/html_parser.py

```diff
@@ -0,0 +1,152 @@
+
+import re
+import six
+
+from bs4 import BeautifulSoup
+
+from .parser import BaseParser
+
+class HtmlParser(BaseParser):
+    def __init__(self, s3_client):
+        super().__init__(s3_client=s3_client)
+        # additional PdfParser constructor code here
+
+    def parse_file(self, html_path):
+        """
+        Extracts text from a html file and returns a string of content.
+        """
+
+        with open(html_path, "rb") as stream:
+            soup = BeautifulSoup(stream, 'lxml')
+
+        # Convert tables to ASCII ones
+        soup = self._replace_tables(soup)
+
+        # Join inline elements
+        soup = self._join_inlines(soup)
+
+        # Make HTML
+        html = ''
+        elements = soup.find_all(True)
+        elements = [el for el in filter(self._visible, elements)]
+        for elem in elements:
+            string = elem.string
+            if string is None:
+                string = self._find_any_text(elem)
+            string = string.strip()
+            if len(string) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)

Reply:
Resolve by using `if string`.